src/sdkAnalytics/getSDKAnalyticsSignature.ts

```diff
@@ -0,0 +1,85 @@
+import {encodeVersion} from "./encodeVersion";
+import {getAnalyticsOptions} from "./getAnalyticsOptions";
+import {ITrackedPropertiesThroughAnalytics} from "./interfaces/ITrackedPropertiesThroughAnalytics";
+import pkg from '../package.json';
+
+/**
+ * @private
+ * @description Try to get the node version out of process, if browser just return 0.0.0
+ */
+function getNodeVersion() {
+  const failedVersion = '0.0.0';
+  if (typeof window !== 'undefined') {
+    return failedVersion;
+  } else {
+    // node env
+    try { return process.versions.node || failedVersion; } catch(e) { return failedVersion; }
+  }
+}
+
+/**
+ * @private
+ * @description Ensure that all values ITrackedPropertiesThroughAnalytics are populated.
+ * Accept a partial map of values and returns the complete interface of ITrackedPropertiesThroughAnalytics
+ * @param {Partial<ITrackedPropertiesThroughAnalytics>} trackedAnalytics
+ * @param {Partial<ITrackedPropertiesThroughAnalytics>} trackedAnalytics
+ */
+function ensureShapeOfTrackedProperties(trackedAnalytics?: Partial<ITrackedPropertiesThroughAnalytics>): ITrackedPropertiesThroughAnalytics {
+  // try to get the process version from node, but if we're on the client return 0.0.0
+  const defaults = {
+    techVersion: getNodeVersion(),
+    sdkCode: 'T', // Base code
+    sdkSemver : pkg.version.split('-')[0], // remove -beta, -alpha etc.
+    responsive: false,
+    placeholder: false,
+    lazyload: false,
+    accessibility: false,
+    urlAnalytics: false
+  };
+
+  if (!trackedAnalytics) {
+    return defaults;
+  } else {
+    return Object.assign({}, defaults, trackedAnalytics);
+  }
+}
+
+
+
+/**
+ * @private
+ * @description Creates the complete SDK signature by using all the values provided by ITrackedPropertiesThroughAnalytics
+ * @return {string} sdkAnalyticsSignature
+ */
+export function getSDKAnalyticsSignature(_trackedAnalytics?: Partial<ITrackedPropertiesThroughAnalytics>): string | 'E' {
```

Comment:
Please add description for function and variables so that a newcomer understands whats going on

Reply:
Done